File: dotnet/src/Microsoft.Agents.Validation/Validators/ContentValidators/RefusalContentValidator.cs
using Microsoft.Agents.Abstractions.Models;

namespace Microsoft.Agents.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for RefusalContent.
/// </summary>
public class RefusalContentValidator : ContentValidatorBase<RefusalContent>
{
    public override ValidationResult Validate(RefusalContent content, ValidationContext? context = null)
    {
        // TODO: Implement specific validation rules for RefusalContent
        return ValidationResult.Success();
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/RefusalContentValidator.ts
import type { RefusalContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';

/**
 * Validator for RefusalContent.
 */
export class RefusalContentValidator extends ContentValidatorBase<RefusalContent> {
  public validate(content: RefusalContent, context?: ValidationContext): ValidationResult {
    // RefusalContent has no strict validation requirements
    return ValidationResult.success();
  }
}
